api/src/api/handlers/pda_worker.rs

```diff
@@ -0,0 +1,93 @@
+use std::str::FromStr;
+
+use crate::{
+    api::handlers::{async_verify::process_verification, is_authorized},
+    db::{
+        models::{extract_instruction, SolanaProgramBuildParams},
+        DbClient,
+    },
+    services::{get_on_chain_hash, onchain::OtterBuildParams},
+    CONFIG,
+};
+use axum::{
+    extract::State,
+    http::{HeaderMap, StatusCode},
+    Json,
+};
+use borsh::BorshDeserialize;
+use serde_json::Value;
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::pubkey::Pubkey;
+use tracing::{error, info, warn};
+
+pub(crate) async fn handle_pda_updates_creations(
+    State(db): State<DbClient>,
+    headers: HeaderMap,
+    Json(payload): Json<Vec<Value>>,
+) -> (StatusCode, &'static str) {
+    info!("Received PDA updates/creation event");
+
+    // Validate authorization
+    if !is_authorized(&headers) {
+        warn!("Unauthorized unverify attempt");
+        return (
+            StatusCode::UNAUTHORIZED,
+            "Missing or invalid authorization header",
+        );
+    }
+
+    // Validate payload
+    let instruction = match extract_instruction(&payload) {
+        Ok(instruction) => instruction,
+        Err(status) => return status,
+    };
+
+    // Process instructions
+    for ix in instruction.instructions {
+        // Only process PDA updates/creations
+        if ix.programId != "verifycLy8mB96wd9wqq3WDXQwM4oU6r42Th37Db9fC" {
+            continue;
+        }
+        let pda_account = &ix.accounts[0];
+        let program_id = &ix.accounts[2];
+
+        let _ = process_pda_upgrade(&db, program_id, pda_account).await;
+    }
+
+    (StatusCode::OK, "PDA updates/creations request received")
+}
+
+async fn process_pda_upgrade(
```

Comment:
If we're checking for updates irrespective of write/update instruction, can we change the name of this method to something more general like `process_otter_verify_instruction`?